options/options.go

```diff
@@ -573,4 +573,8 @@ func UnsetEnv() {
 		_ = os.Unsetenv(opt.Env)
 		_ = os.Unsetenv(strings.TrimPrefix(opt.Env, envPrefix))
 	}
+
+	// Unset the Kaniko environment variable which we set it in the
+	// Dockerfile to ensure correct behavior during building.
+	_ = os.Unsetenv("KANIKO_DIR")
```

Comment:
Review: This was left in a built image, but not in a cached image. I chose to go this path, but it also means it's not as "easy" to re-envbuilder inside a built image. But I didn't think that's _really_ a use-case, wdyt @johnstcn?

Reply:
I think the main use-case of a cached image is to run the lifecycle scripts. We don't ever want to delete the filesystem or run kaniko inside a pre-built image anyway, so I don't see a reason to need this env.

Reply:
This isn't actually relevant for the cached image, since it doesn't set this env. It's only for a built image. But I'd argue the same logic applies there.